vicon_nexus_unity_stream_py/_main.py

```diff
@@ -2,248 +2,100 @@
 
 """Main script."""
 
-import sys
 import json
-import msgpack
 import pandas as pd
 from pathlib import Path
 from datetime import datetime
 
-from flask import Flask, send_file, make_response
-from flask_restful import Resource, Api
+from fastapi import FastAPI, WebSocket
 from loguru import logger
-from alive_progress import alive_bar
+
+from vicon_nexus_unity_stream_py._utils import process_return_value
 
 try:
     from vicon_dssdk import ViconDataStream
 except ImportError:
     logger.error("Make sure vicon DataStreamSDK is installed: Follow the instructions in https://www.vicon.com/software/datastream-sdk/\n")
     #raise
 
-def get_client(connection=None):
+app = FastAPI()
+CLIENT = None
+DEFAULT_HOST = '0.0.0.0'
+DEFAULT_PORT = 5001
+
+#parameters
+TICK_RATE = 0.01
+
+@app.websocket("/marker/")
+async def websocket_endpoint(websocket: WebSocket):
+    global CLIENT
+    await websocket.accept()
+
+    while CLIENT.IsConnected and CLIENT.GetFrame():
+        data = get_data(CLIENT, "marker")
+        await websocket.send_json(data)
+        await asyncio.sleep(TICK_RATE)  
+
+
+def get_vicon_instance(connection=None):
+    """Returns the vicon client instance."""
     if connection is None:
         connection = 'localhost:801'
     client = ViconDataStream.Client()
 
-    logger.info('Connecting...')
     while not client.IsConnected():
         client.Connect(connection)
-    logger.info('Connected to vicon data stream')
+    client.SetStreamMode(ViconDataStream.Client.StreamMode.EClientPullPreFetch)
```

Comment:
Oh A sidenote: Do you want to experiment with Pull, Push and PullPrefetch to see if this is a factor for dropped frames in unity

Reply:
but other than that
Saul Goodman 👍

Reply:
good point. We'll do that in a sperate PR.